test/com/google/enterprise/adaptor/CommandStreamParserTest.java

```diff
@@ -167,7 +167,6 @@ public void testSupportedDelimiterCharacters() throws Exception {
   public void testRetriever() throws IOException {
     String source = "GSA Adaptor Data Version 1 [\n]\n"
         + "id=123\n"
-        + "up-to-date\n"
```

Comment:
There's content buried at the end of this input string, and you can't have both (since they correspond to calling respondNotModified and getOutputStream). I picked the content here to preserve the other things this test was checking, and added a separate test for up-to-date, and a test for the two commands conflicting, and one more for trying to do something else you can't do at the end (similar to setAcl).